src/Uno.UI/UI/Xaml/Controls/ViewBox/Viewbox.cs

```diff
@@ -55,13 +57,26 @@ protected override Size MeasureOverride(Size availableSize)
 
 		protected override Size ArrangeOverride(Size finalSize)
 		{
+			if(finalSize.Width == 0 || finalSize.Height == 0)
+			{
+				return default;
+			}
+
 			var (scaleX, scaleY) = GetScale(finalSize, _container.DesiredSize);
 
-			_container.RenderTransform = new ScaleTransform()
+			if (Math.Abs(scaleX - 1d) < 0.001d && Math.Abs(scaleY - 1d) < 0.001d)
 			{
-				ScaleX = scaleX,
-				ScaleY = scaleY
-			};
+				_container.RenderTransform = null;
+			}
+			else
+			{
+				var transform = _container.RenderTransform as ScaleTransform ?? new ScaleTransform();
+				transform.ScaleX = scaleX;
+				transform.ScaleY = scaleY;
+				transform.CenterX = finalSize.Width / 2d;
+				transform.CenterY = finalSize.Height / 2d;
```

Comment:
You could have set the `RenderTransformOrigin` is there a specific reason to do it like that?